docs(CONTRIBUTING.md): update steps for adding a new extension

qiaofenlin opened the pull request:
### Fix broken Dockerfile link in CONTRIBUTING.md

**Description:**  
I noticed that the [Dockerfile link](https://github.com/supabase/postgres/blob/develop/Dockerfile) in the `CONTRIBUTING.md` file was broken (404). I have attempted to fix this issue in this pull request.

**Reason for Major Changes:**  
While fixing the link, I found that there are multiple versions of Dockerfiles in the current project. Therefore, I updated the documentation to reflect this and to help contributors add extensions to the appropriate Dockerfile(s).


**Screenshots:**  
![image](https://github.com/user-attachments/assets/9444af12-4a17-4c43-b191-28dac8f6d76a)

**Notes:**  
If there are any problems with my changes, please feel free to point them out.  
Or, feel free to make further modifications based on this PR. Thank you!

qiaofenlin committed:
docs(CONTRIBUTING.md): update steps for adding a new extension

samrose commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,19 @@
 
-Extensions can either be built from source or installed through a debian package. In general, you want to add the installation commands for your extension to the [Dockerfile](Dockerfile) following the steps below.
+Supabase Postgres supports multiple Dockerfiles for different versions and deployment scenarios (e.g., `Dockerfile-15`, `Dockerfile-17`, `Dockerfile-kubernetes`, `Dockerfile-orioledb-17`).  
+**Please add your extension to each relevant Dockerfile according to the images you want to support.**
```
```suggestion

```

Suggesting removing this line, because in the Dockerfile per major version, the installation of postgres and extensions happens as a "bundle" and there is already logic in the packaging that handles which packages should target which major version. Instead, if you want to target a major version with an extension, this needs to happen in the nix files.

samrose commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,19 @@
-5. Copy your extension's package from build stage to [extensions stage](Dockerfile#L851).
+### Steps
+
+1. **Add a build stage** for your extension in the target Dockerfile, named after your extension.  
```
All of these steps 1-4 are no longer accurate, as we no longer handle build/install of extension in Dockerfiles. 

Instead, the instructions are now here for adding extensions: https://github.com/supabase/postgres/blob/develop/nix/docs/adding-new-package.md

samrose reviewed (changes requested):
@qiaofenlin thanks for your great contribution! I left a few notes on on the PR. The contributing is very out of date. Definitely the change you submitted about supporting multiple Dockerfiles is good and accurate. The rest is superceded by Supabase use of Nix. I think we'll include your contribution on the Dockerfile variations, but we need to also update the rest on how to contribute extensions.

qiaofenlin committed:
docs: update extension addition instructions in CONTRIBUTING.md

qiaofenlin commented:
@samrose Thank you for your review and suggestions!  

I have updated the documentation according to your feedback and addressed the issues you pointed out.
If there’s anything else that needs adjustment, please let me know. Thanks again for your help!

samrose reviewed (changes requested):
We should also remove all these lines since we don't do any of this in Dockerfiles any more

> Here's a minimal example:
> 
> ```dockerfile
> ARG pg_graphql_release=1.1.0
> 
> ####################
> # 19-pg_graphql.yml
> ####################
> FROM base as pg_graphql
> # Download package archive
> ARG pg_graphql_release
> ADD "https://github.com/supabase/pg_graphql/releases/download/v${pg_graphql_release}/pg_graphql-v${pg_graphql_release}-pg${postgresql_major}-${TARGETARCH}-linux-gnu.deb" \
>     /tmp/pg_graphql.deb
> 
> ####################
> # Collect extension packages
> ####################
> FROM scratch as extensions
> COPY --from=pg_graphql /tmp/*.deb /tmp/
> ```
> 
> Using this process maximises the effectiveness of Docker layer caching, which significantly speeds up our CI builds.

samrose commented:
If you remove those lines, I'll approve and we'll expand on it from there, as there is more to know, but I don't expect anyone outside of Supabse to yet know all of the other contributing info based on recent changes.

qiaofenlin committed:
docs: remove pg_graphql extension Dockerfile example from CONTRIBUTING.md

qiaofenlin commented:
> If you remove those lines, I'll approve and we'll expand on it from there, as there is more to know, but I don't expect anyone outside of Supabse to yet know all of the other contributing info based on recent changes.

@samrose Thank you for your feedback!
I've removed those lines as suggested. If there are any other changes or improvements needed, please let me know—I'm happy to update the PR further as required.

qiaofenlin commented:
Hi  @samrose, 
I have addressed the review suggestions and pushed the corresponding changes some time ago. I noticed that the review has not been updated for a while, so I wanted to kindly check if there are any remaining concerns or if there’s anything else I should improve.

Please let me know if you need any further information from my side. Thank you for your time and feedback!

Best regards.

qiaofenlin commented:
> If you remove those lines, I'll approve and we'll expand on it from there, as there is more to know, but I don't expect anyone outside of Supabse to yet know all of the other contributing info based on recent changes.

Hi @samrose ,
Just a gentle follow-up regarding this PR.
I have addressed all the review suggestions and pushed the changes.
If there are any remaining issues or concerns, please let me know—I'm happy to make further adjustments if needed.
If everything looks good, I would appreciate your approval or any next steps.
Thank you for your time and support!
Best regards,
